src/Keboola/StorageApi/Options/SearchTablesOptions.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Keboola\StorageApi\Options;
+
+class SearchTablesOptions
+{
+    /** @var string|null */
+    private $metadataKey;
+
+    /** @var string|null */
+    private $metadataValue;
+
+    /** @var string|null */
+    private $metadataProvider;
+
+    /**
+     * @param string|null $metadataKey
+     * @param string|null $metadataValue
+     * @param string|null $metadataProvider
+     * @return SearchTablesOptions
+     */
+    public static function create($metadataKey, $metadataValue, $metadataProvider)
+    {
+        $self = new self();
+        $self->metadataKey = $metadataKey;
+        $self->metadataValue = $metadataValue;
+        $self->metadataProvider = $metadataProvider;
+        return $self;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getMetadataKey()
+    {
+        return $this->metadataKey;
+    }
+
+    /**
+     * @param string|null $metadataKey
+     * @return SearchTablesOptions
+     */
+    public function setMetadataKey($metadataKey)
+    {
+        $this->metadataKey = $metadataKey;
+        return $this;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getMetadataValue()
+    {
+        return $this->metadataValue;
+    }
+
+    /**
+     * @param string|null $metadataValue
+     * @return SearchTablesOptions
+     */
+    public function setMetadataValue($metadataValue)
+    {
+        $this->metadataValue = $metadataValue;
+        return $this;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getMetadataProvider()
+    {
+        return $this->metadataProvider;
+    }
+
+    /**
+     * @param string|null $metadataProvider
+     * @return SearchTablesOptions
+     */
+    public function setMetadataProvider($metadataProvider)
+    {
+        $this->metadataProvider = $metadataProvider;
+        return $this;
+    }
+
+    public function validate()
+    {
+        foreach ($this->toArray() as $option) {
+            if (!empty($option)) {
+                return;
+            }
+        }
+
+        throw new \Exception('At least one option must be set');
```

Comment:
Client\Exception ?

Reply:
Toť právě otázka, viz můj komentář, teoreticky by to měla být nějaká `ValidationException` nebo to pustit do api bez validace.

Reply:
jako bylo by fajn, to nepouštět vůbec na API, ale nemáme jenom 1 klienta a ty ostatní tu validaci dělat nebudou. Nechal bych to opravdu na straně aplikace, kde ti API vrátí, že si to blbě poskládal. Já nad tímhle klientam přemýšlím jako nad úplně dummy věcí, která jenom dělá requesty. Pak si řekneme, že jsou povinný 2 věci místo jedný a to budeš muset upravovat na 2 místech